src/sentry/api/bases/organization_integrations.py

```diff
@@ -35,7 +36,56 @@ def validate_integration_id(integration_id):
             raise Http404
 
     @staticmethod
-    def get_organization_integration(organization, integration_id) -> RpcOrganizationIntegration:
+    def get_organization_integration(
+        organization_id: int, integration_id: int
+    ) -> OrganizationIntegration:
+        """
+        Get just the cross table entry.
+        Note: This will still return organization integrations that are pending deletion.
+
+        :param organization:
+        :param integration_id:
+        :return:
+        """
+        try:
+            return OrganizationIntegration.objects.get(
+                integration_id=integration_id,
+                organization_id=organization_id,
+            )
+        except OrganizationIntegration.DoesNotExist:
+            raise Http404
+
+    @staticmethod
+    def get_integration(organization_id: int, integration_id: int) -> Integration:
+        """
+        Note: The integration may still exist even when the
+        OrganizationIntegration cross table entry has been deleted.
```

Comment:
Some (but not all) integration records are shared across orgs which is why there is this junction table.